base/locks.jl

```diff
@@ -136,41 +194,45 @@ unsafe_convert(::Type{Ptr{Void}}, m::Mutex) = m.handle
 
 function _uv_hook_close(x::Mutex)
     h = x.handle
-    x.handle = C_NULL
-    ccall(:uv_mutex_destroy, Void, (Ptr{Void},), h)
-    Libc.free(h)
-    nothing
+    if h != C_NULL
+        x.handle = C_NULL
+        ccall(:uv_mutex_destroy, Void, (Ptr{Void},), h)
+        Libc.free(h)
+        nothing
+    end
 end
 
-function lock!(m::Mutex)
+function lock(m::Mutex)
```

Comment:
ReadWriteLocks.jl, Celeste.jl, and Dtree.jl were all using this name. Not great to be renaming things right now.